_posts/2020-02-03-apache-beam-how-beam-runs-on-top-of-flink.md

```diff
@@ -0,0 +1,157 @@
+---
+layout: post
+title: 'Apache Beam: How Beam Runs on Top of Flink'
+date: 2020-01-30T12:00:00.000Z
+authors:
+- maximilian:
+  name: "Maximilian Michels"
+  twitter: "stadtlegende"
+- markos:
+  name: "Markos Sfikas"
+  twitter: "MarkSfik"
+excerpt: This blog post, discusses the reasons to use Flink together with Beam for your stream processing needs and takes a closer look on how Flink works with Beam under the hood.
+
+---
+
+Note: This blog post is based on the talk ["Beam on Flink: How Does It Actually Work?"](https://www.youtube.com/watch?v=hxHGLrshnCY).
+
+[Apache Flink](https://flink.apache.org/) and [Apache Beam](https://beam.apache.org/) are open-source frameworks for parallel, distributed data processing at scale. Unlike Flink, Beam does not come with a full-blown execution engine of its own but plugs into other execution engines like Apache Flink, Apache Spark, or Google Cloud Dataflow. In this blog post, we discuss the reasons to use Flink together with Beam for your stream processing needs. We also take a closer look on how Flink works with Beam to provide an idea of the technical aspects of running Beam pipelines with Flink. We hope you find some useful information on how and why the two frameworks can be utilized in combination. For more information you can refer to the corresponding [documentation](https://beam.apache.org/documentation/runners/flink/) on the Beam website or contact the community through the [Beam mailing list](https://beam.apache.org/community/contact-us/).
+
+
+# What is Apache Beam
+
+[Apache Beam](https://beam.apache.org/) is an open-source, unified model for defining batch and streaming data-parallel processing pipelines. It is unified in the sense that you use a single API, in contrast to using a seperate API for batch and streaming like it is the case in Flink. Beam was originally developed by Google which released it in 2014 as the Cloud Dataflow SDK. In 2016, it was donated to [the Apache Software Foundation](https://www.apache.org/) with the name of Beam. It has been developed by the open-source community ever since. With Apache Beam, developers can write data processing jobs, also known as pipelines, in multiple languages, e.g. Java, Python, Go, SQL. A pipeline is then executed by one of Beam’s Runners. A Runner is responsible for translating Beam pipelines such that they can run on an execution engine. Every supported execution engine has a Runner. The following Runners are available: Apache Flink, Apache Spark, Apache Samza, Hazelcast Jet, Google Cloud Dataflow, and others.
```

Comment:
```suggestion
[Apache Beam](https://beam.apache.org/) is an open-source, unified model for defining batch and streaming data-parallel processing pipelines. It is unified in the sense that you use a single API, in contrast to using a separate API for batch and streaming like it is the case in Flink. Beam was originally developed by Google which released it in 2014 as the Cloud Dataflow SDK. In 2016, it was donated to [the Apache Software Foundation](https://www.apache.org/) with the name of Beam. It has been developed by the open-source community ever since. With Apache Beam, developers can write data processing jobs, also known as pipelines, in multiple languages, e.g. Java, Python, Go, SQL. A pipeline is then executed by one of Beam’s Runners. A Runner is responsible for translating Beam pipelines such that they can run on an execution engine. Every supported execution engine has a Runner. The following Runners are available: Apache Flink, Apache Spark, Apache Samza, Hazelcast Jet, Google Cloud Dataflow, and others.
```